spec/model_uri_spec.rb

```diff
@@ -477,6 +477,12 @@
       expect(u1.canonicalize!.to_s).to eq u2.to_s
       expect(u1).to eq u2
     end
+    it "#canonicalize does not fail with Encoding::CompatibilityError on weird IRIs" do
+      u1 = RDF::URI "htЫtp://user:passoЫd@exaЫmple.com:8080/path ПУТЬ?queЫry=valЫue#fragmeЫnt"
+      u2 = RDF::URI "ht%D0%ABtp://user:passoЫd@exaЫmple.com:8080/path%20ПУТЬ?queЫry=valЫue#fragmeЫnt"
+      expect(u1.canonicalize.to_s.dup.force_encoding(u2.to_s.encoding)).to eq u2.to_s
```

Comment:
@ujifgc Can you comment on the `#dup` call, here?

I was able to work around the problems in `#normalized_authority` for Ruby 2.3, but wound up chasing some other problems here. `#dup` looks unnecessary, since `#canonicalize` already dups the whole `URI` instance; but I wanted to make sure I'm not missing something.